src/functions/sendCandidateResults/application/service/fault-provider.ts

```diff
@@ -32,54 +53,60 @@ export interface IFaultProvider {
 export class FaultProvider implements IFaultProvider {
 
   public getDrivingFaults(testData: TestData | undefined, category: string): Fault[] {
-    let faults: Fault[] = [];
-
     switch (category) {
       case TestCategory.B:
-        faults = getDrivingFaultsCatB(testData as CatBUniqueTypes.TestData);
-        break;
+        return getDrivingFaultsCatB(testData as CatBUniqueTypes.TestData);
       case TestCategory.BE:
-        faults = getDrivingFaultsCatBE(testData as CatBEUniqueTypes.TestData);
-        break;
+        return getDrivingFaultsCatBE(testData as CatBEUniqueTypes.TestData);
+      case TestCategory.C:
+        return getDrivingFaultsCatC(testData as CatCUniqueTypes.TestData);
+      case TestCategory.C1:
+        return getDrivingFaultsCatC1(testData as CatC1UniqueTypes.TestData);
+      case TestCategory.CE:
+        return getDrivingFaultsCatCE(testData as CatCEUniqueTypes.TestData);
+      case TestCategory.C1E:
+        return getDrivingFaultsCatC1E(testData as CatC1EUniqueTypes.TestData);
       default:
-        faults = getDrivingFaultsCatB(testData as CatBUniqueTypes.TestData);
-        break;
+        return getDrivingFaultsCatB(testData as CatBUniqueTypes.TestData);
```

Comment:
I've highlighted this kind of issue already on a few different PRs... I don't think that defaulting to a cat B. is the right thing to do in those kinds of operations and we should stop doing this - I'd prefer to see an error being thrown here and handled elsewhere rather than pretending that we're dealing with cat B. as a default which can cause more of headache than it's worth 

I know that you haven't introduced that this pattern - if you think that it would take more time to get this cone - I'm happy to extract this kind of refactor to a separate piece of work/PR (just not to block this unnecessarily)